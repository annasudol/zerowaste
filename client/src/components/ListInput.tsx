import { Form, Input, Button, Select, InputNumber } from 'antd';
import { FormInstance } from 'antd/lib/form';
import { AutoComplete, List } from "../components"
import * as React from "react";

const tailLayout = {
    wrapperCol: { offset: 8, span: 16 },
};
const formRef = React.createRef<FormInstance>();


const validateMessages = {
    required: '${label} is required!',
    number: {
        range: '${label} must be between ${min} and ${max}',
    },
};

// tslint:disable-next-line: no-empty-interface
interface ListInputProps {
    placeholder?: string
    setDetailedIngredients(value: string[]): void
    form(detailedIngredients: string[]): void
    detailedIngredients: string[] | []
}
export const ListInput: React.FunctionComponent<ListInputProps> = ({ setDetailedIngredients, detailedIngredients, form }): any => {
    const [input, setInput] = React.useState<string>("")

    const handleIngredientChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        event.preventDefault()

        setInput(event.target.value)

    };

    const handleIngredientsChange = () => {
        setDetailedIngredients([...detailedIngredients, input]);
        formRef.current?.resetFields();
        setInput("");
    };

    React.useEffect(() => {
        form(detailedIngredients)
    }, [detailedIngredients])

    return (
        <>
            <Input onChange={handleIngredientChange} value={input} />
            <Button type="dashed" onClick={handleIngredientsChange} disabled={input.length < 4}>Add Ingredient</Button>
        </>


    );
}
