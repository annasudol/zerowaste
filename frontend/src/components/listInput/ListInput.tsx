import { Input, Button } from 'antd';
import { FormInstance } from 'antd/lib/form';
import React, { FC, ReactElement } from 'react';
const formRef = React.createRef<FormInstance>();

interface ListInputProps {
    placeholder?: string
    setDetailedIngredients(value: string[]): void
    form(detailedIngredients: string[]): void
    detailedIngredients: string[]
}

export const ListInput: FC<ListInputProps> = ({ setDetailedIngredients, detailedIngredients, form }): ReactElement => {
    const [input, setInput] = React.useState<string>('')

    const handleIngredientChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setInput(event.target.value)
    };

    const handleIngredientsChange = () => {
        setDetailedIngredients([...detailedIngredients, input]);
        formRef.current?.resetFields();
        setInput('');
    };

    React.useEffect(() => {
        form(detailedIngredients)
    }, [detailedIngredients, form])

    return (
        <>
            <Input onChange={handleIngredientChange} value={input} />
            <Button type='dashed' onClick={handleIngredientsChange} disabled={input.length < 4}>Add Ingredient</Button>
        </>
    );
}
