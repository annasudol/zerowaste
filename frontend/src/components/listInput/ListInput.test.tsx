import { shallow } from 'enzyme';
import React from 'react';
import { ListInput } from './ListInput';
import toJson from 'enzyme-to-json';
import { Input, Button } from 'antd';


describe('component', (): void => {
  describe('ListInput', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(
        <ListInput detailedIngredients={[]} setDetailedIngredients={jest.fn()} form={jest.fn()} />,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('responds for handleIngredientsChange', () => {
        const mockSetDetailedIngredients = jest.fn()
        const wrapper = shallow(<ListInput form={jest.fn()} detailedIngredients={['test1']} setDetailedIngredients={mockSetDetailedIngredients} />);
        wrapper.find(Input).prop<Function>('onChange')({ target: { value: 'test2'}});
        wrapper.find(Button).simulate('click');
        expect(mockSetDetailedIngredients).toHaveBeenCalledWith(['test1', 'test2']);
      });
  });
});