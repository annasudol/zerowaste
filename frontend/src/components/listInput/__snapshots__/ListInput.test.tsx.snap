// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component ListInput match snapshot 1`] = `
<Fragment>
  <Input
    onChange={[Function]}
    type="text"
    value=""
  />
  <Button
    block={false}
    disabled={true}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="dashed"
  >
    Add Ingredient
  </Button>
</Fragment>
`;
